### PR TITLE
fix(TDP-4961) Integration test with DI jobs

### PR DESCRIPTION
**Pull request by aleboulanger**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4961

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/199/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/199/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/200/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/200/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/201/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/201/)

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/DatasetStep.java [hunk 1]
```diff
@@ -91,6 +81,71 @@ public void thenICheckSemanticTypeExist(String semanticTypeId, String columnId)
...
+
+        }
+        context.storeDatasetRef(datasetId, suffixedName);
+    }
+
+    @Given("^I have a dataset with parameters:$")
+    public void iHaveADatasetWithParameters(DataTable dataTable) throws Throwable {
+        Map<String, String> parameters = new HashMap<>(dataTable.asMap(String.class, String.class));
+
+        // in case of only name parameter, we should use a suffixed dataSet name
```
[the remark this answers is not in the document]
Comment (aleboulanger) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/DatasetStep.java]: i will update it in TDP-5335. this method is used both for tDatasetInput and tDatasetOutput.
for tDatasetOutput feature, we suffix the datasetName so we have to do the same when we check the existence of the dataset after the execution of the job

**Pull request merged**